holoviews/core/dimension.py

```diff
@@ -1334,10 +1335,14 @@ def __init__(self, items=None, identifier=None, parent=None, **kwargs):
     @classmethod
     def _process_items(cls, vals):
         "Processes list of items assigning unique paths to each."
+        from .layout import AdjointLayout
+
         if type(vals) is cls:
             return vals.data
-        elif not isinstance(vals, (list, tuple)):
+        elif isinstance(vals, (AdjointLayout, str)):
```

Comment:
Thanks for the PR! Could you please clarify why you need to check for the `str` case?

Reply:
I'm also curious about this...I'm sure I knew why this might happen at some point!

Reply:
I don't have a great reason for adding `str` to this check other than this is how it worked before by converting `"ABC"` to `["ABC"]` instead of `["A", "B", "C"]`. Could _possible_ give a better traceback by giving the whole string instead of only the first letter.

Reply:
I ran the whole test suite storing the type of `vals` each time this method is called, the unique types recorded are: `{"<class 'holoviews.core.layout.AdjointLayout'>", "<class 'list'>"}`.

Reply:
I understand this better now: definitely needs a comment to explain!